contracts/script/deploy/sepolia/DeploymentInitializer.sol

```diff
@@ -0,0 +1,202 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity =0.8.12;
+
+import {IRegistryCoordinator, RegistryCoordinator} from "lib/eigenlayer-middleware/src/RegistryCoordinator.sol";
+import {IPauserRegistry} from
+    "lib/eigenlayer-middleware/lib/eigenlayer-contracts/src/contracts/interfaces/IPauserRegistry.sol";
+import {IStakeRegistry} from "lib/eigenlayer-middleware/src/interfaces/IStakeRegistry.sol";
+import {IEigenDAThresholdRegistry, EigenDAThresholdRegistry} from "src/core/EigenDAThresholdRegistry.sol";
+import {IEigenDARelayRegistry, EigenDARelayRegistry} from "src/core/EigenDARelayRegistry.sol";
+import {PaymentVault} from "src/payments/PaymentVault.sol";
+import {IEigenDADisperserRegistry, EigenDADisperserRegistry} from "src/core/EigenDADisperserRegistry.sol";
+import {EigenDAServiceManager} from "src/core/EigenDAServiceManager.sol";
+import {IRewardsCoordinator} from
+    "lib/eigenlayer-middleware/lib/eigenlayer-contracts/src/contracts/interfaces/IRewardsCoordinator.sol";
+import {ProxyAdmin, TransparentUpgradeableProxy} from "@openzeppelin/contracts/proxy/transparent/ProxyAdmin.sol";
+import "./DeploymentTypes.sol";
+
+/**
```

Comment:
just for understanding - the intent of this contract is for providing an on-chain initialization step where dependency injection happens as a precursor tx which allows a deployer to then verify before actually invoking the deploy? 

curious about the following:
1. is this only used for bootstrapping/ genesis deployment of the contracts or will it also be leveraged for future upgrades? If for bootstrapping could we not achieve the same effect through a forge deployment script which verifies non-eigenda contract dependencies through as set of invariants before beginning the initialization steps for each eigenda contract?

Reply:
Your understanding is basically correct. There's no reason this framework cannot be used for upgrades too.